### ISSUE TITLE
CLRRuntimeHost.GetCurrentAppDomainId fails

### ISSUE THREAD

**Issue by aparamon** [dotnet/coreclr#7597]
To reproduce:

1) Build coreclr.dll.
> build.cmd

2) Build mscoree.tlb, e.g. by MIDL compiler:
> midl /I ..\pal\prebuilt\inc\ /D FEATURE_CORECLR mscoree.idl

3) Execute the following code that starts the CLR runtime and outputs default appdomain ID (adjust coreclr.dll path to match your build dir and architecture):

```
import comtypes
import comtypes.client
from ctypes import *
import os.path

mscoree = comtypes.client.GetModule('mscoree.tlb')

Runtime = POINTER(mscoree.ICLRRuntimeHost)()
GetCLRRuntimeHost = oledll.LoadLibrary(os.path.join(coreclr.git', 'trunk', 'bin', 'Product', 'Windows_NT.x64.Debug', 'coreclr')).GetCLRRuntimeHost
GetCLRRuntimeHost(pointer(comtypes.GUID('{712AB73F-2C22-4807-AD7E-F501D7B72C2D}')), byref(Runtime));

print(Runtime.Start())
print(Runtime.GetCurrentAppDomainId())
```

The script finishes with unhandled exception:
_ctypes.COMError: (-2147467263, 'Не поддерживается', (None, None, None, 0, None))
-2147467263 is E_NOTIMPL

The expected is the ID of default appdomain.

**Comment by jkotas** [comment]
The recomended way to host CoreCLR is via the flat C APIs defined in 
https://github.com/dotnet/coreclr/blob/master/src/coreclr/hosts/inc/coreclrhost.h.

The COM hosting interfaces are left-over from full .NET Framework. They do not work well in CoreCLR.

**Comment by jkotas** [comment]
cc @janvorli

**Comment by aparamon** [comment]
Thank you for your suggestion!
Is it expected that some future version of MS .NET Framework will also have these C APIs?

**Comment by jkotas** [comment]
I agree that it would be useful to have a common hosting interface. There are no concrete plans for it yet - it will take a while.

**Comment by gkhanna79** [comment]
@aparamon Are there any open items this issue is tracking?

CC @rahku
